de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/invoicecandidates/impl/InvoicingParamsObject.java

```diff
@@ -0,0 +1,310 @@
+package de.metas.rest_api.invoicecandidates.impl;
+
+/*
+ * #%L
+ * de.metas.swat.base
+ * %%
+ * Copyright (C) 2015 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+import javax.annotation.Nullable;
+
+import de.metas.invoicecandidate.api.IInvoicingParams;
+import lombok.ToString;
+
+@ToString
+public class InvoicingParamsObject implements IInvoicingParams
```

Comment:
what's the difference between this class and PlainInvoicingParams ?

Reply:
When I started the implementation, I thought it would have some fields different from PlainInvoicingParams. I will remove it.

Reply:
Also, I didn't know exactly in which flows is the PlainInvoicingParams class used and as I understood, I needed updateLocationAndContactForInvoice field to be set and in PlainInvoicingParams it was only a Getter annotation, so....I thought it's better not to interfere and create a new one. I have replaced it now.

Reply:
AFAIU you added now added `updateLocationAndContactForInvoice` to PlainInvoicingParams, so it's OK